src/LiveComponent/src/ComponentWithMultiStepFormTrait.php

```diff
@@ -0,0 +1,315 @@
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent;
+
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\UX\LiveComponent\Attribute\LiveAction;
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\LiveComponent\Storage\StorageInterface;
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+use Symfony\UX\TwigComponent\Attribute\PostMount;
+
+use function Symfony\Component\String\u;
+
+/**
+ * Trait for managing multistep forms in Symfony UX LiveComponent.
+ *
+ * This trait simplifies the implementation of multistep forms by handling
+ * step transitions, form validation, data persistence, and state management.
+ * It provides a structured API for developers to integrate multistep forms
+ * into their components with minimal boilerplate.
+ *
+ * @author Silas Joisten <silasjoisten@proton.me>
+ * @author Patrick Reimers <preimers@pm.me>
+ * @author Jules Pietri <heahdude@yahoo.fr>
+ */
+trait ComponentWithMultiStepFormTrait
+{
+    use ComponentWithFormTrait;
+    use DefaultActionTrait;
+
+    #[LiveProp]
+    public ?string $currentStepName = null;
+
+    /**
+     * @var string[]
+     */
+    #[LiveProp]
+    public array $stepNames = [];
+
+    /**
+     * Checks if the current form has validation errors.
+     */
+    public function hasValidationErrors(): bool
+    {
+        return $this->form->isSubmitted() && !$this->form->isValid();
+    }
+
+    /**
+     * @internal
+     *
+     * Initializes the form and restores the state from storage.
+     *
+     * This method must be executed after `ComponentWithFormTrait::initializeForm()`.
+     */
+    #[PostMount(priority: -250)]
+    public function initialize(): void
+    {
+        $this->currentStepName = $this->getStorage()->get(\sprintf('%s_current_step_name', self::prefix()), $this->formView->vars['current_step_name']);
+
+        $this->form = $this->instantiateForm();
+
+        $formData = $this->getStorage()->get(\sprintf('%s_form_values_%s', self::prefix(), $this->currentStepName));
+
+        $this->form->setData($formData);
+
+        $this->formValues = [] === $formData
+            ? $this->extractFormValues($this->getFormView())
+            : $formData;
+
+        $this->stepNames = $this->formView->vars['steps_names'];
+
+        // Do not move this. The order is important.
+        $this->formView = null;
+    }
+
+    /**
+     * Advances to the next step in the form.
+     *
+     * Validates the current step, saves its data, and moves to the next step.
+     * Throws a RuntimeException if no next step is available.
+     */
+    #[LiveAction]
+    public function next(): void
+    {
+        $this->submitForm();
+
+        if ($this->hasValidationErrors()) {
+            return;
+        }
```

Comment:
If `!form->isValid`, then a `UnprocessableEntityHttpException` is triggered in the componentwithformtrait, no ? So as i see it, the hasValidationErrors() check is redondant (i may have missed something)

 ```suggestion
        $this->submitForm();
```

Reply:
not sure to be honest. i can test it.